public/pages/CreateMonitor/containers/DefineMonitor/DefineMonitor.js

```diff
@@ -279,7 +283,7 @@ class DefineMonitor extends Component {
       content = (
         <ExtractionQuery
           response={JSON.stringify(response || '', null, 4)}
-          isDarkMode={this.isDarkMode}
+          isDarkMode={this.context.isDarkMode}
```

Comment:
any reason why you pass other core services (`notifications`, `httpClient`) via props, but get `isDarkMode` via context here instead?

Reply:
I guess the same question goes for all places where you are using context instead of props.

Reply:
I see, makes sense. Personally, I think for overall style it would probably be better to do one or the other - if some services need to be passed deeply, I think a good change could be removing the props and consuming everything via context like what you've done for `isDarkMode`. More info [here](https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#how-can-i-avoid-passing-core-services-deeply-within-my-ui-component-tree).

Could be a separate PR later to try to standardize this, just my thoughts

Reply:
You are right, the style should be consistent. As all the `httpClient` has been passed by "props", I have changed all the `notifications` and `isDarkMode` to be passed by "props". 😂. Btw, the AD related code in this repo is using the Context, and that's a reference to me.